manim/scene/scene_file_writer.py

```diff
@@ -388,15 +381,17 @@ def write_frame(self, frame_or_renderer: np.ndarray | OpenGLRenderer):
         elif config.renderer == RendererType.CAIRO:
             frame = frame_or_renderer
             if write_to_movie():
-                self.writing_process.stdin.write(frame.tobytes())
+                av_frame = av.VideoFrame.from_ndarray(frame, format="rgba")
+                for packet in self.video_stream.encode(av_frame):
+                    self.video_container.mux(packet)
             if is_png_format() and not config["dry_run"]:
                 self.output_image_from_array(frame)
 
     def write_opengl_frame(self, renderer: OpenGLRenderer):
         if write_to_movie():
-            self.writing_process.stdin.write(
-                renderer.get_raw_frame_buffer_object_data(),
-            )
+            av_frame = av.VideoFrame.from_ndarray(renderer.get_frame(), format="rgba")
```

Comment:
As I mentioned in another comment, I think we should take the `n_frames: int = 1` option to this method and add the same multiple times **after** we converted it into a `av.VideoFrame` and got the packets (this is a list, so it does not exhaust). I did not measure any significant speed-up yet, but it does not make sense to me to do the conversion multiple times when you could avoid it very easily.